### ISSUE TITLE
Error using RefineDet with cpp_classification example of Caffe

### ISSUE THREAD

**Issue by alexandre2r** [sfzhang15/RefineDet#104]
Hi, first of all congrats for the repo!
I am trying to use the RefineDet pretrained in COCO dataset with the classification example in C++ provided by Caffe (main):
`./main deploy.prototxt coco_refinedet_vgg16_320x320_final.caffemodel mean.binaryproto labels.txt cat.jpg`
But I am having a problem with the last layer (backtrace):
```
F0810 09:01:07.707479 21292 main.cpp:83] Check failed: labels_.size() == output_layer->channels() (80 vs. 1) Number of labels is different from the output layer dimension.
*** Check failure stack trace: ***
    @     0x7f038d8a85cd  google::LogMessage::Fail()
    @     0x7f038d8aa433  google::LogMessage::SendToLog()
    @     0x7f038d8a815b  google::LogMessage::Flush()
    @     0x7f038d8aae1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x404d29  Classifier::Classifier()
    @           0x4064d5  main
    @     0x7f038cf5b830  __libc_start_main
    @           0x4045d9  _start
    @              (nil)  (unknown)
```
Any suggestion? Thanks

**Comment by sfzhang15** [comment]
@alexandre2r 
I am not familiar with this operation. Could you explain why you want to use this opertation?

**Comment by alexandre2r** [comment]
@sfzhang15 I am using this example to know if it is possible to run the RefineDet in C++ because I did not found a demo on C++ in the repo, only on Python. Is it right?
Thanks

**Comment by sfzhang15** [comment]
@alexandre2r 
You can refer [this](https://github.com/weiliu89/caffe/blob/ssd/examples/ssd/ssd_detect.cpp) to implement C++ inference.

**Comment by alexandre2r** [comment]
Thank you @sfzhang15 . I was using the wrong the .cpp for detection from the Caffe master instead of the one from SSD. This way it works perfectly.

**Comment by newforrestgump001** [comment]
@alexandre2r Could you give me some hints of c++ inference code?
I used the ssd_detect.bin and shows the following message:
[2020-05-22 15:17:55.839] [caffe] [info] Attempting to upgrade batch norm layers using deprecated params: deploy.prototxt
[2020-05-22 15:17:55.839] [caffe] [info] Successfully upgraded batch norm layers using deprecated params.
[2020-05-22 15:17:55.840] [caffe] [info] Initializing net from parameters: 
[2020-05-22 15:17:55.841] [caffe] [error] Unknown bottom blob 'data' (layer 'stem1/conv', bottom index 0.000000)
terminate called after throwing an instance of 'CaffeErrorException'
